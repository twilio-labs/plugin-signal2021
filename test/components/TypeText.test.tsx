import { render } from 'ink-testing-library';
import * as React from 'react';
import {
  getTextFrames,
  TypeText,
} from '../../src/components/animated/TypeText';

describe('getTextFrames', () => {
  it('should create frames from text', () => {
    const actual = getTextFrames('Hello');
    const expected = ['', 'H', 'He', 'Hel', 'Hell', 'Hello'];
    expect(actual).toEqual(expected);
  });
});

describe('<TypeText />', () => {
  it('should display text', () => {
    const rendered = render(<TypeText text="Hi there!" />);
    expect(rendered.lastFrame()).toEqual('Hi there!');
  });

  it('should match snapshot', () => {
    const rendered = render(<TypeText text="Hi there!" />);
    expect(rendered.lastFrame()).toMatchSnapshot();
  });
});
