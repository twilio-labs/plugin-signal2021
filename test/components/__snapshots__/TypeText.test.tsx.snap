// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TypeText /> should match snapshot 1`] = `"Hi there!"`;
